cmd/kubeadm/app/preflight/checks.go

```diff
@@ -909,15 +909,9 @@ func (MemCheck) Name() string {
 
 // InitNodeChecks returns checks specific to "kubeadm init"
 func InitNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfiguration, ignorePreflightErrors sets.Set[string], isSecondaryControlPlane bool, downloadCerts bool) ([]Checker, error) {
-	if !isSecondaryControlPlane {
-		// First, check if we're root separately from the other preflight checks and fail fast
-		if err := RunRootCheckOnly(ignorePreflightErrors); err != nil {
-			return nil, err
-		}
-	}
-
 	manifestsDir := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName)
 	checks := []Checker{
+		IsPrivilegedUserCheck{},
```

Comment:
it should be added only for this condition is true
```
if !isSecondaryControlPlane {
```
just create the checks variable earlier:
var checks []Checker